### PR TITLE
Some error corrections and some enhancement done

### PR DESCRIPTION
**Pull request by crocri**
- Ding event is now working again (ring Interkom)
- auto livestream creation takes now value from config, before fix
- snapshot now async, because snapshot and livestream in parallel does not work
- livestream duration now settable via tree entry
  will be auto reset via livestream request. After request is done, duration time is reset to config value
- two new config entries: auto_livestream and auto_snapshot - can be disabled now if someone wants to react manually
- some minor corrections to code

### CODE REVIEW

**Review by mcm1957 — COMMENTED** [review]
Please remove line
```"installedFrom": "iobroker-community-adapters/ioBroker.ring"```
from package.json.

Codechanges should be reviewed by @theimo1221

**Review by theimo1221 — CHANGES_REQUESTED** [review]
Files in `build/` directory are generated during `npm run build` so all of your changes in that folder would be overwritten during next changes.
So you have to perform your changes within `src/` folder in the respective TypeScript file.

File: build/lib/ownRingCamera.js [hunk 1]
```diff
@@ -418,26 +444,30 @@ class OwnRingCamera extends ownRingDevice_1.OwnRingDevice {
...
-        }
-        this.silly(`Would have recorded due to "${EventState[state]}", but we are already reacting.`);
-        if (uuid) {
-            setTimeout(() => {
-                this.debug(`delayed uuid recording`);
-                this.takeSnapshot(uuid);
-            }, 23000);
-        }
-    }
+       if (this._autoSnapshot || this._autoLiveStream) {
```
Comment (theimo1221) [inline build/lib/ownRingCamera.js:447]: Inverting this statement you would reduce nesting, which improves code readability.

**Comment by crocri** [comment]
- Removed "installedFrom": "iobroker-community-adapters/ioBroker.ring" (Martin, you mean io-package.json, right?)
- adapted ts files (but I'm not really qualified to do this Thiemo ;-)
- if expression inverted, correct Thiemo? if (a || b) -> if (!(a && b)), or do you mean anything else? If, do it by your own, please.

**Comment by theimo1221** [comment]
@crocri Thanks for modification permission and your changes.

image

Did you open/editted the code in an IDE like VsCode?
Additionaly running `npm run build` results in multiple compilation issues.

I'll see if I can fix those quickly

**Comment by crocri** [comment]
Nope, heard the first time from VsCode (Microsoft). Never developed an iobroker adapter, only look into the build code directly. Changes done and tested in my ioBroker system only.

**Comment by theimo1221** [comment]
Good Morning @crocri,

could you test the newest version within your system?

In general IDEs (Webstorm would be the one I personally use) are programs which help you to write/edit/inspect/run code.
Like in this case, installing the npm packages, performing the build process, marking errors/mistakes, giving auto-completion, etc. So if you want to dig deeper into this space, I'd recommend trying e.g. VsCode as it is free.

**Comment by crocri** [comment]
ioBroker dev (VM) and VS Code installed, "new" ring adapter compiled and tested. No issues left (after debugging ;-)

**Comment by theimo1221** [comment]
Thanks @crocri will review this later today

**Comment by theimo1221** [comment]
Later on a deep dive into your ownRingDevice changes is necessary and a functional test is pending, afterwards we could proceed.

**Pull request merged**